woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -2538,3 +2520,78 @@ def test_ltype_conversions_nullable_types():
         "Please confirm the underlying data is consistent with logical type Integer."
     with pytest.raises(TypeConversionError, match=error_msg):
         df.ww.set_types({'int_null': 'Integer'})
+
+
+def test_init_with_partial_schema_infer_types(sample_df):
+    test_df = sample_df.copy()
+    sample_df.ww.init()
+    schema = sample_df.ww[['id', 'full_name']].ww.schema
+
+    assert test_df.ww.schema is None
+    test_df.ww.init_with_partial_schema(schema)
+    assert isinstance(test_df.ww.schema, TableSchema)
+    assert_schema_equal(test_df.ww.schema, sample_df.ww.schema)
+
+
+def test_init_with_partial_schema_full_schema(sample_df):
+    test_df = sample_df.copy()
+    sample_df.ww.init()
+
+    assert test_df.ww.schema is None
+    test_df.ww.init_with_partial_schema(sample_df.ww.schema)
+    assert isinstance(test_df.ww.schema, TableSchema)
+    assert_schema_equal(test_df.ww.schema, sample_df.ww.schema)
+
+
+def test_init_with_partial_schema_override_schema(sample_df):
+    test_df = sample_df.copy()
+    sample_df.ww.init()
+    schema = sample_df.ww[['integer', 'categorical']].ww.schema
+
+    assert test_df.ww.schema is None
+    test_df.ww.init_with_partial_schema(schema,
+                                        logical_types={'categorical': 'Unknown'},
+                                        use_standard_tags={'id': False})
+    assert sample_df.ww.logical_types['categorical'] == schema.logical_types['categorical']
+    assert test_df.ww.logical_types['categorical'] != schema.logical_types['categorical']
+    assert test_df.ww.logical_types['integer'] == schema.logical_types['integer']
+    assert not test_df.ww.semantic_tags['id']
+
+
+def test_init_with_partial_schema_metadata_deep_copy(sample_df):
+    test_df = sample_df.copy()
+    sample_df.ww.init(table_metadata={'test': '123'})
+    schema = sample_df.ww[['id', 'full_name']].ww.schema
+    assert schema.metadata
+    assert test_df.ww.schema is None
+    test_df.ww.init_with_partial_schema(schema)
+    assert test_df.ww.schema.metadata
+    test_df.ww._schema.metadata = {'test2': '456'}
+    assert schema.metadata != test_df.ww.schema.metadata
```

Comment:
I think this test wouldn't catch the deepcopy behavior that we need to avoid.

Setting `test_df.ww._schema.metadata` with a completely new dictionary object (line 2569)  will only also change `schema`'s metadata if `schema` and `test_df.ww._schema` are the same `TableSchema` object.

This test needs to update the `test_df.ww._schema.metadata` dictionary that's already there. Adding a new entry will, I believe, test if a shallow copy _of the dictionary_ was performed (if both metadata dicts are the same object). But to confirm that a deep copy has occurred, we'll need to update a mutable value in the metadata dictionary (both lists, for example, would update if the lists are the same object). 

And example of this kind of test is `test_dataframe_methods_on_accessor_new_schema_object` or `test_iloc_table_does_not_propagate_changes_to_data`

Reply:
you're right!  I've changed it to copy the same things that `_get_subset_schema` copies

Reply:
@tuethan1999 This test still needs to be updated to not just add a new element to the metadata dictionary, but to update the value of an element already in the dictionary. 

```python
sample_df.ww.init(table_metadata={'test': [1,2]})

...

test_df.ww._schema.metadata['test'].append(3)
```

That way it's checking that a deep copy has occured